### PR TITLE
support for remote VOIs

### PR DESCRIPTION
**Pull request by naylor-b**

### CODE REVIEW

File: openmdao/core/driver.py [hunk 1]
```diff
@@ -106,19 +108,115 @@ def _setup_driver(self, problem):
...
 
-        self._responses = model.get_responses(recurse=True)
-        self._objs = model.get_objectives(recurse=True)
-        self._cons = model.get_constraints(recurse=True)
+        con_set = set()
+        obj_set = set()
+        dv_set = set()
+
+        self._remote_dvs = dv_dict = {}
+        self._remote_cons = con_dict = {}
```
Comment (Kenneth-T-Moore) [inline openmdao/core/driver.py]: Looks like a few vars to add to the attributes in the class docstring.

Comment (naylor-b) [reply openmdao/core/driver.py]: fixed

File: openmdao/core/driver.py [hunk 2]
```diff
@@ -106,19 +117,115 @@ def _setup_driver(self, problem):
...
+            for i, vname in enumerate(model._var_allprocs_abs_names['output']):
+                owner = owning_ranks[vname]
+                if vname in dv_set:
+                    dv_dict[vname] = (owner, sizes[owner, i])
+                if vname in con_set:
+                    con_dict[vname] = (owner, sizes[owner, i])
+                if vname in obj_set:
+                    obj_dict[vname] = (owner, sizes[owner, i])
+
+        self._remote_responses = self._remote_cons.copy()
```
Comment (hwangjt) [inline openmdao/core/driver.py]: Missed `self._remote_responses` in the attributes section of the class docstring

Comment (naylor-b) [reply openmdao/core/driver.py]: fixed

**Pull request merged**